### ISSUE TITLE
train loss some wrong 

### ISSUE THREAD

**Issue by 10183308** [MaybeShewill-CV/CRNN_Tensorflow#167]
INFO: 10-09 19:55:21: train_shadownet.py:139 * 140230364735232 Training from scratch
INFO: 10-09 19:55:25: train_shadownet.py:174 * 140230364735232 Epoch: 1 cost= 197.760651
INFO: 10-09 19:55:26: train_shadownet.py:174 * 140230364735232 Epoch: 2 cost= 172.005493
INFO: 10-09 19:55:26: train_shadownet.py:174 * 140230364735232 Epoch: 3 cost= 142.324768
INFO: 10-09 19:55:26: train_shadownet.py:174 * 140230364735232 Epoch: 4 cost= 101.206573
INFO: 10-09 19:55:26: train_shadownet.py:174 * 140230364735232 Epoch: 5 cost= 75.101547
INFO: 10-09 19:55:26: train_shadownet.py:174 * 140230364735232 Epoch: 6 cost= 62.528458
INFO: 10-09 19:55:26: train_shadownet.py:174 * 140230364735232 Epoch: 7 cost= 43.640678
INFO: 10-09 19:55:26: train_shadownet.py:174 * 140230364735232 Epoch: 8 cost= 36.773048
INFO: 10-09 19:55:27: train_shadownet.py:174 * 140230364735232 Epoch: 9 cost= 33.266392
INFO: 10-09 19:55:27: train_shadownet.py:174 * 140230364735232 Epoch: 10 cost= 33.096649
INFO: 10-09 19:55:27: train_shadownet.py:174 * 140230364735232 Epoch: 11 cost= 30.440775
INFO: 10-09 19:55:27: train_shadownet.py:174 * 140230364735232 Epoch: 12 cost= 29.101267
INFO: 10-09 19:55:27: train_shadownet.py:174 * 140230364735232 Epoch: 13 cost= 31.529339
INFO: 10-09 19:55:27: train_shadownet.py:174 * 140230364735232 Epoch: 14 cost= 27.663685
INFO: 10-09 19:55:28: train_shadownet.py:174 * 140230364735232 Epoch: 15 cost= 27.336868
INFO: 10-09 19:55:28: train_shadownet.py:174 * 140230364735232 Epoch: 16 cost= 26.076229
INFO: 10-09 19:55:28: train_shadownet.py:174 * 140230364735232 Epoch: 17 cost= 29.231804
INFO: 10-09 19:55:28: train_shadownet.py:174 * 140230364735232 Epoch: 18 cost= 28.733067
INFO: 10-09 19:55:28: train_shadownet.py:174 * 140230364735232 Epoch: 19 cost= 31.574839
INFO: 10-09 19:55:28: train_shadownet.py:174 * 140230364735232 Epoch: 20 cost= 25.050362
INFO: 10-09 19:55:29: train_shadownet.py:174 * 140230364735232 Epoch: 21 cost= 26.840332
INFO: 10-09 19:55:29: train_shadownet.py:174 * 140230364735232 Epoch: 22 cost= 26.538689
INFO: 10-09 19:55:29: train_shadownet.py:174 * 140230364735232 Epoch: 23 cost= 31.059734
INFO: 10-09 19:55:29: train_shadownet.py:174 * 140230364735232 Epoch: 24 cost= 28.754152
INFO: 10-09 19:55:29: train_shadownet.py:174 * 140230364735232 Epoch: 25 cost= 27.408020
INFO: 10-09 19:55:29: train_shadownet.py:174 * 140230364735232 Epoch: 26 cost= 27.259018
INFO: 10-09 19:55:29: train_shadownet.py:157 * 140230364735232 Cost didn't improve beyond 0.001000 for 7 epochs, stopping early.
Done

Could you help me how to solve it ?

thank you very

**Comment by MaybeShewill-CV** [comment]
@10183308 set early stop to false in config/global_config.py file

**Comment by 10183308** [comment]
Mean accuracy is 0.000000
INFO: 10-09 20:11:20: train_shadownet.py:169 * 140110462555904 Epoch: 75 cost= 26.064211 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:21: train_shadownet.py:169 * 140110462555904 Epoch: 76 cost= 26.023987 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:22: train_shadownet.py:169 * 140110462555904 Epoch: 77 cost= 24.003937 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:23: train_shadownet.py:169 * 140110462555904 Epoch: 78 cost= 26.151737 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:25: train_shadownet.py:169 * 140110462555904 Epoch: 79 cost= 27.721252 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:26: train_shadownet.py:169 * 140110462555904 Epoch: 80 cost= 26.118002 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:27: train_shadownet.py:169 * 140110462555904 Epoch: 81 cost= 27.228586 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:28: train_shadownet.py:169 * 140110462555904 Epoch: 82 cost= 24.062744 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:29: train_shadownet.py:169 * 140110462555904 Epoch: 83 cost= 24.330078 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:30: train_shadownet.py:169 * 140110462555904 Epoch: 84 cost= 23.941357 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:32: train_shadownet.py:169 * 140110462555904 Epoch: 85 cost= 26.703520 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:33: train_shadownet.py:169 * 140110462555904 Epoch: 86 cost= 28.245827 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:34: train_shadownet.py:169 * 140110462555904 Epoch: 87 cost= 24.811604 seq distance=  1.000000 train accuracy=  0.000000
Mean accuracy is 0.000000
INFO: 10-09 20:11:35: train_shadownet.py:169 * 140110462555904 Epoch: 88 cost= 23.889328 seq distance=  1.000000 train accuracy=  0.000000



my loss is aways 100% ,why

**Comment by MaybeShewill-CV** [comment]
@10183308 The model can not converge that fast

**Comment by 10183308** [comment]
@MaybeShewill-CV 
Do you mean this is normal?

the loss now is very large ,but mu accuracy is aways 0.000

**Comment by ryanwcyin** [comment]
@10183308 It’s normal if you have diverse and large dataset. The cost will decrease.

**Comment by 10183308** [comment]
@MaybeShewill-CV  OK,I will continue to train my model ,thank you very much.

**Comment by MaybeShewill-CV** [comment]
@10183308 You are welcome

**Comment by hustzxl** [comment]
> @MaybeShewill-CV OK,I will continue to train my model ,thank you very much.

is  the  accuracy  normal now?I have  the same  question except my loss is low  .but  the  accuracy  is always  0 ,the  distance  is 1, and  I print the pred  found  they all is  empty

**Comment by hustzxl** [comment]
@10183308    
is the accuracy normal now?I have the same question except my loss is low .but the accuracy is always 0 ,the distance is 1, and I print the pred found they all is empty

**Comment by bluseking** [comment]
after 40000 epoches  the loss can not be reduced,
INFO: 12-08 06:50:44: train_shadownet.py:172 * 140718831200000 Epoch: 39984 cost= 86.564514
INFO: 12-08 06:50:45: train_shadownet.py:172 * 140718831200000 Epoch: 39985 cost= 82.274956
INFO: 12-08 06:50:45: train_shadownet.py:172 * 140718831200000 Epoch: 39986 cost= 82.426605
INFO: 12-08 06:50:46: train_shadownet.py:172 * 140718831200000 Epoch: 39987 cost= 89.131454
INFO: 12-08 06:50:46: train_shadownet.py:172 * 140718831200000 Epoch: 39988 cost= 89.673813
INFO: 12-08 06:50:47: train_shadownet.py:172 * 140718831200000 Epoch: 39989 cost= 82.546646
INFO: 12-08 06:50:47: train_shadownet.py:172 * 140718831200000 Epoch: 39990 cost= 88.471100
INFO: 12-08 06:50:47: train_shadownet.py:172 * 140718831200000 Epoch: 39991 cost= 86.229065
INFO: 12-08 06:50:48: train_shadownet.py:172 * 140718831200000 Epoch: 39992 cost= 77.373032
INFO: 12-08 06:50:48: train_shadownet.py:172 * 140718831200000 Epoch: 39993 cost= 86.483749
INFO: 12-08 06:50:49: train_shadownet.py:172 * 140718831200000 Epoch: 39994 cost= 81.790146
INFO: 12-08 06:50:49: train_shadownet.py:172 * 140718831200000 Epoch: 39995 cost= 80.195297
INFO: 12-08 06:50:50: train_shadownet.py:172 * 140718831200000 Epoch: 39996 cost= 80.802933
INFO: 12-08 06:50:50: train_shadownet.py:172 * 140718831200000 Epoch: 39997 cost= 84.123375
INFO: 12-08 06:50:51: train_shadownet.py:172 * 140718831200000 Epoch: 39998 cost= 84.217491
INFO: 12-08 06:50:51: train_shadownet.py:172 * 140718831200000 Epoch: 39999 cost= 80.480850
INFO: 12-08 06:50:52: train_shadownet.py:172 * 140718831200000 Epoch: 40000 cost= 87.768478
Done 
are there some problems on the model？

**Comment by TianboChen** [comment]
@bluseking I have the same problem with you,do you solve the problem?

**Comment by kei6** [comment]
@bluseking @TianboChen  I have the same problem here, did you solve it?

**Comment by kei6** [comment]
@TianboChen It doesn't solve my problem, 0.1 is the default learning rate in config.py. I've already used that learning rate and the loss is high (~20), the prediction is only one character for every sample or empty.

**Comment by Antonio-hi** [comment]
@TianboChen @kei6 @bluseking are you solve the problem? Can anyone help me

**Comment by kei6** [comment]
@Antonio-hi I have fixed my problem by using another repo. Before change repo, try to use small learning-rate (i use 0.0001). Training CTC loss needs to converge at 0.xxx to work!

**Comment by TianboChen** [comment]
@kei6 The model I trained using this repo is not working well on other real data.The training CTC Loss is 0.xxx.Which new repo do you use?

**Comment by kei6** [comment]
@TianboChen Oh, CTC loss is 0.xxx but not work. That's weird! Let try the model from this repo [this repo](https://github.com/xiaofengShi/CHINESE-OCR/blob/master/train/keras-train/model.py). They're using GRU, I got the char-match accuracy of 0.82, 0.45 for full-match.
